pkg/pillar/cmd/domainmgr/containerd.go

```diff
@@ -90,70 +94,159 @@ func ctrRun(domainName, xenCfgFilename, imageHash string, envList map[string]str
 // but instead use containerd to set it up, and xl to run it, so this does not need to do anything.
 // If/when we invert it, and have containerd launch the container with a microvm wrapper,
 // *then* this will need to work.
-func ctrStop(id string, force bool) error {
-	log.Infof("ctrStop %s %t\n", id, force)
+func ctrStop(containerID string, force bool) error {
+	log.Infof("ctrStop %s %t\n", containerID, force)
 	log.Infof("ctr stop done\n")
 	return nil
 }
 
 // ctrRm remove an existing container
-func ctrRm(id string) error {
-	log.Infof("ctrRm %s\n", id)
+func ctrRm(containerID string) error {
+	log.Infof("ctrRm %s\n", containerID)
+	ctrdClient, err := getContainerdClient(ctrdSocket)
+	if err != nil {
+		log.Errorf("Could not build containerd client. %v", err.Error())
+		return fmt.Errorf("Could not build containerd client. %v", err.Error())
+	}
+	defer ctrdClient.Close()
+	ctrdCtx := getContainerdContext(CtrdServicesNamespace)
+
+	container, err := loadContainer(containerID)
+	if err != nil {
+		return err
+	}
+	task, err := container.Task(ctrdCtx, cio.NewAttach(cio.WithStdio))
+	if err != nil {
+		log.Errorf("Could not fetch task of container: %v. %v", containerID, err.Error())
+		return fmt.Errorf("Could not fetch task of container: %v. %v", containerID, err.Error())
+	}
+	_, err = task.Delete(ctrdCtx)
+	if err != nil {
+		log.Errorf("Unable to delete task of container: %v. %v", containerID, err.Error())
+		return fmt.Errorf("Unable to delete task of container: %v. %v", containerID, err.Error())
+	}
+	err = container.Delete(ctrdCtx)
+	if err != nil {
+		log.Errorf("Unable to delete container: %v. %v", containerID, err.Error())
+		return fmt.Errorf("Unable to delete container: %v. %v", containerID, err.Error())
+	}
 	log.Infof("ctrRm done\n")
 	return nil
 }
 
 // ctrCreate create a new container but do not start it
-func ctrCreate(id, imageHash string) error {
-	return nil
+func ctrCreate(containerID, imageName string, status *types.DomainStatus) (string, error) {
```

Comment:
Hmm. I see the only thing we are using `status` for in this func is to get the env vars, which is a `map[string]string`. Could we just pass the `status.EnvVariables` to this func as a `map[string]string`? That would make this func entirely eve-independent, and really easy to test.

Reply:
And yes, I am acutely aware that I passed it through in the previous PR. It is your good work here that made me see the possibilities for improving it. :-)

Reply:
Done.